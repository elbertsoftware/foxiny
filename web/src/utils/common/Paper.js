import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import MuiPaper from '@material-ui/core/Paper';
import { capitalize } from '@material-ui/core/utils/helpers';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  backgroundLight: {
    backgroundColor: '#fcfcfc',
  },
  backgroundMain: {
    backgroundColor: theme.palette.secondary.light,
  },
  backgroundDark: {
    backgroundColor: theme.palette.secondary.dark,
  },
  root: {
    maxWidth: 935,
    margin: 'auto',
    marginTop: '15px',
    marginBottom: '10px',
  },
  padding: {
    padding: theme.spacing.unit,
  },
});

function Paper(props) {
  const { background, classes, className, padding, ...other } = props;

  return (
    <MuiPaper
      elevation={2}
      className={classNames(
        classes[`background${capitalize(background)}`],
        {
          [classes.padding]: padding,
        },
        classes.root,
        className,
      )}
      {...other}
    />
  );
}

Paper.propTypes = {
  background: PropTypes.oneOf(['light', 'main', 'dark']),
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  padding: PropTypes.bool,
};

Paper.defaultProps = {
  background: 'main',
  padding: false,
};

export default withStyles(styles)(Paper);
