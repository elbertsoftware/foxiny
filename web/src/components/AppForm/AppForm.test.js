import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import TestAppForm from './AppForm';
import LayoutBody from '../LayoutBody/LayoutBody';
import Paper from '../Paper/Paper';

const defaultProps = {
    square: false,
    width: 'small',
  },
  AppForm = props => <TestAppForm {...defaultProps} {...props} />;

describe('App form', () => {
  it('render correctly App Form component', () => {
    // Snapshot
    const AppFormComponent = renderer.create(<AppForm />).toJSON();
    expect(AppFormComponent).toMatchSnapshot();
  });
  describe('render correctly children', () => {
    it('chilren is undefined by default', () => {
      const AppFormComponent = mount(<AppForm />);
      expect(AppFormComponent.props().children).toEqual(undefined);
    });
    it('render with defined children', () => {
      const AppFormComponent = shallow(<AppForm>Minh Kha</AppForm>);
      expect(AppFormComponent.contains('Minh Kha')).toBeTruthy();
    });
  });
  describe('render correctly width, square props', () => {
    it('check default props by default values', () => {
      const AppFormComponent = shallow(<AppForm />);
      expect(AppFormComponent.props().width).toBe('small');
    });
    it('render correctly with LayoutBody, and Paper', () => {
      // Snapshot
      const AppFormComponent = renderer.create(<AppForm />);
      const AppFormInstance = AppFormComponent.root;
      expect(AppFormInstance.findByType(LayoutBody).props.width).toBe('small');
      expect(AppFormInstance.findByType(Paper).props.square).toBeFalsy();
    });
    it('props is render with defined width, and square', () => {
      const props = {
        width: 'medium',
        square: true,
      };
      const AppFormComponent = mount(<AppForm {...props} />);
      expect(AppFormComponent.prop('width')).toBe('medium');
      expect(AppFormComponent.prop('square')).toBeTruthy();
    });
  });
});
