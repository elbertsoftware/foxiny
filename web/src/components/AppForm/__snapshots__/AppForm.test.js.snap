// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App form render correctly App Form component 1`] = `
<div
  className="AppForm-root-1"
>
  <div
    className="LayoutBody-widthSmall-5 LayoutBody-margin-3 LayoutBody-marginBottom-4"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation2 Paper-backgroundLight-11 Paper-root-14 AppForm-paper-2 MuiPaper-rounded"
    />
  </div>
</div>
`;
